NuRadioReco/framework/base_trace.py

```diff
@@ -248,35 +248,36 @@ def deserialize(self, data_pkl):
 
     def add_to_trace(self, channel):
         """
-        Adds the trace of another channel (with potentially diffrent length and _trace_start_time) to
-        the trace of this channel without changing the sampling. If used on an empty trace with a defined
-        self._sampling_rate and self.self._trace_start_time, and self._time_trace containing zeros, this
-        function can be seen as recording the channel in a specific readout window.
+        Adds the trace of another channel to the trace of this channel. The trace is only added within the 
+        time window of "this" channel.
+        If this channel is an empty trace with a defined _sampling_rate and _trace_start_time, and a 
+        _time_trace containing zeros, this function can be seen as recording a channel in the specified
+        readout window.
 
         Parameters
         ----------
         channel: BaseTrace
-            Channel to add to current channel
+            The channel whose trace is to be added to the trace of this channel.
         """
 
         assert self.get_number_of_samples() is not None, "No trace is set for this channel"
         assert self.get_sampling_rate() == channel.get_sampling_rate(), "Sampling rates of the two channels do not match"
 
         tt_readout = self.get_times()
         t0_readout = self.get_trace_start_time()
-        t1_readout = t0_readout +  self.get_number_of_samples() / self.get_sampling_rate()
+        t1_readout = tt_readout[-1]
         sampling_rate_readout = self.get_sampling_rate()
         n_samples_readout = self.get_number_of_samples()
 
         tt_channel = channel.get_times()
         t0_channel = channel.get_trace_start_time()
-        t1_channel = t0_channel + channel.get_number_of_samples() / channel.get_sampling_rate()
+        t1_channel = tt_channel[-1]
         sampling_rate_channel = channel.get_sampling_rate()
         n_samples_channel = channel.get_number_of_samples()
 
         # We handle 1+2x2 cases:
-        # 1. Channel is completely outside readout window:
-        if t1_channel < t0_readout or t1_readout < t0_channel:
+        # 1. Channel is completely outside readout window (less than one full bin overlap):
+        if t1_channel < tt_readout[1] or t1_readout < t0_channel:
```

Comment:
why was to old condition not okay?

Reply:
I actually don't think I need this change anymore after fixing the other bug. I'll run a test to see of the old condition causes problems.

Reply:
What was the bug, I do not see related changes.

Reply:
There were no problems in my test so I reverted this change now.